### PR TITLE
feat: simple filter transformer nc

### PR DESCRIPTION
**Pull request by LauJohansson**
Tranformer for making a simple filtering on one column for one value.

### CODE REVIEW

File: src/spetlr/transformers/simple_dataframe_filter_transformer_nc.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        The non-consuming version of DataFrameFilterTransformer()
+        without a default column name.
+
+        A simple helper class to filter a dataframe based on the passed col_name
+        if its value equals to the specified col_value.
+
+        Args:
+            col_value: The value of the column to filter the dataframe
+            col_name: The name of the column to filter the dataframe. The default
+                is 'messageType' as this class is mostly used to filter dataframes
```
Comment (Okami1) [inline src/spetlr/transformers/simple_dataframe_filter_transformer_nc.py:28]: Det er fra den originale version

**Comment by MartinBoge** [comment]
We could also have a transformer that simply accepts a costume filter statement to make it more flexible.

**Comment by LauJohansson** [comment]
> We could also have a transformer that accepts a custom filter statement to make it more flexible.

Good point. I think it is a balance between how much customization is needed as input to a transformer.

Should we have a simple filtertransformer like this + a transformer that can have a filter-string as input?

Or, should they be merged?

**Review by Okami1 — APPROVED** [review]

**Comment by MartinBoge** [comment]
> > We could also have a transformer that accepts a custom filter statement to make it more flexible.
> 
> Good point. I think it is a balance between how much customization is needed as input to a transformer.
> 
> Should we have a simple filtertransformer like this + a transformer that can have a filter-string as input?
> 
> Or, should they be merged?

My 2c is that we should have a simple transformer to accept an argument "query: str" because it gives more flexibility. I.e.:
```python
from typing import List, Union

from pyspark.sql import DataFrame
from spetlr.etl import TransformerNC


class FilterQueryTransformer(TransformerNC):
    def __init__(
        self,
        query: str,
        dataset_input_keys: Union[str, List[str]] = None,
        dataset_output_key: str = None,
    ):
        super().__init__(
            dataset_input_keys=dataset_input_keys,
            dataset_output_key=dataset_output_key,
        )
        self.query = query

    def process(self, df: DataFrame) -> DataFrame:
        print("Filter transformer on sql query")
        return df.filter(self.query)
```

**Comment by Okami1** [comment]
> > We could also have a transformer that accepts a custom filter statement to make it more flexible.
> 
> Good point. I think it is a balance between how much customization is needed as input to a transformer.
> 
> Should we have a simple filtertransformer like this + a transformer that can have a filter-string as input?
> 
> Or, should they be merged?

We actually already have a version of what Martin is talking about on IIS in the "filterQueryTransformer"

**Pull request merged**